apps/web/src/components/borrow/modal/DepositModal.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import {
+  BoxLayout,
+  Button,
+  CenterLayout,
+  Icon,
+  Modal,
+  StackLayout,
+  Text,
+} from '@bloxifi/ui'
+import { BorrowAndLending, Tokens } from '@bloxifi/core'
+import { CheckAllowanceFunction } from '@bloxifi/types'
+import { useTranslation } from 'react-i18next'
+import { ethers } from 'ethers'
+import { Formik } from 'formik'
+import * as Yup from 'yup'
+
+import { TableInput } from '../table/TableInput'
+import { TransactionOverview } from '../table/TransactionOverview'
+
+import { Web3Container } from '@/containers/Web3Container'
+import { ReservesData } from '@/containers/WalletContainer'
+
+interface Props {
+  /**
+   * Boolean value that defines if modal is open or closed
+   */
+  isOpen?: boolean
+  /**
+   * Function that defines closing of modal
+   */
+  onClose?: () => void
+  /**
+   * Selected asset reserve data
+   */
+  reserveData?: ReservesData
+}
+
+const initailReserveData = {
+  balance: undefined,
+  icon: '',
+  fullName: '',
+  supplyAPY: undefined,
+  variableBorrowAPY: undefined,
+  id: '',
+  name: null,
+  symbol: null,
+  decimals: undefined,
+  totalATokenSupply: undefined,
+  totalCurrentVariableDebt: undefined,
+  liquidityRate: undefined,
+  variableBorrowRate: undefined,
+  underlyingAsset: '',
+}
+
+export const DepositModal = ({
+  isOpen,
+  onClose,
+  reserveData = initailReserveData,
```

Comment:
can you remind me why it's not working like this
`reserveData = {} as ReserveData` :)